cmd/thanos/downsample.go

```diff
@@ -55,21 +57,22 @@ func runDownsample(
 	httpAddr string,
 	dataDir string,
 	gcsBucket string,
+	s3Config *s3.Config,
 	syncDelay time.Duration,
+	component string,
 ) error {
-	gcsClient, err := storage.NewClient(context.Background())
+
+	bkt, closeFn, err := client.NewBucket(&gcsBucket, *s3Config, reg, component)
```

Comment:
are we able to move it down to 
```
// Start cycle of syncing blocks from the bucket and garbage collecting the bucket.
{
 // here
}
```
block? The scope of it is limited to that right?

Reply:
ignore me here. Lets do same as here: https://github.com/improbable-eng/thanos/pull/313#discussion_r186973760